### PR TITLE
Add DETIC perception for OVMM task

### PR DESCRIPTION
**Pull request by yvsriram**
## Motivation and Context
@mukulkhanna's changes for adding DETIC 
1. Loads categories from episodes
2. DETIC predictions for categories
3. visualizer updates for DETIC

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: projects/habitat_ovmm/README.md [hunk 1]
```diff
@@ -107,17 +107,25 @@ python projects/habitat_ovmm/eval_vectorized.py
 Results are saved to `datadump/images/eval_floorplanner/`.
 
 ## Install Detic
-TODO Fix these instructions to start by downloading submodule
+
 ```
```
Comment (exhaustin) [inline projects/habitat_ovmm/README.md:111]:
```suggestion
```sh
```

File: projects/habitat_ovmm/README.md [hunk 2]
```diff
@@ -107,17 +107,25 @@ python projects/habitat_ovmm/eval_vectorized.py
...
-cd /path/to/home-robot-dev/src/home_robot/home_robot/agent/perception/detection/detic
-python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'
+git submodule update --init --recursive src/third_party/detectron2 src/home_robot/home_robot/perception/detection/detic/Detic
+cd src/third_party/detectron2
+pip install -r requirements.txt
+pip install -e ./
+cd ../../..
```
Comment (exhaustin) [inline projects/habitat_ovmm/README.md:116]:
```suggestion
pip install src/third_party/detectron2/requirements.txt
pip install -e src/third_party/detectron2
```

File: projects/habitat_ovmm/README.md [hunk 3]
```diff
@@ -107,17 +107,25 @@ python projects/habitat_ovmm/eval_vectorized.py
...
 pip install -r requirements.txt
+
 mkdir models
 wget https://dl.fbaipublicfiles.com/detic/Detic_LCOCOI21k_CLIP_SwinB_896b32_4x_ft4x_max-size.pth -O models/Detic_LCOCOI21k_CLIP_SwinB_896b32_4x_ft4x_max-size.pth
 
 # Test it with
 wget https://web.eecs.umich.edu/~fouhey/fun/desk/desk.jpg
 python demo.py --config-file configs/Detic_LCOCOI21k_CLIP_SwinB_896b32_4x_ft4x_max-size.yaml --input desk.jpg --output out.jpg --vocabulary lvis --opts MODEL.WEIGHTS models/Detic_LCOCOI21k_CLIP_SwinB_896b32_4x_ft4x_max-size.pth
+
+cd ../../../../../../..
```
Comment (exhaustin) [inline projects/habitat_ovmm/README.md:128]:
```suggestion
cd -
```

**Review by exhaustin — APPROVED** [review]
Looks good in general, some minor suggestions on making the instructions cleaner.

File: src/home_robot_sim/home_robot_sim/env/habitat_objectnav_env/constants.py [hunk 4]
```diff
@@ -486,6 +486,158 @@ def num_sem_categories(self):
...
+    v: k for k, v in rearrange_detic_categories_indexes.items()
+}
+
+
+rearrange_detic_categories_legend_path = str(
+    Path(__file__).resolve().parent / "rearrange_detic_categories_legend.png"
+)
+
+
+class RearrangeDETICCategories(SemanticCategoryMapping):
```
Comment (cpaxton) [inline src/home_robot_sim/home_robot_sim/env/habitat_objectnav_env/constants.py:563]: Some comments describing what this is for, would be useful

**Review by cpaxton — APPROVED** [review]
lgtm - once tests pass

File: projects/habitat_ovmm/eval_vectorized.py [hunk 5]
```diff
@@ -251,7 +249,7 @@ def stop():
     config, config_str = get_config(args.habitat_config_path, opts=args.opts)
     OmegaConf.set_readonly(config, True)
     baseline_config = OmegaConf.load(
-        "projects/habitat_ovmm/configs/agent/floorplanner_eval.yaml"
+        "projects/habitat_ovmm/configs/agent/floorplanner_detic_eval.yaml"
```
Comment (cpaxton) [inline projects/habitat_ovmm/eval_vectorized.py:252]: We should add a command line flag for this?

Comment (yvsriram) [reply projects/habitat_ovmm/eval_vectorized.py:252]: There already exists one. Using it now. Updated command in README

**Pull request merged**